brickschema/validate.py

```diff
@@ -0,0 +1,341 @@
+"""
+The `validate` module implements a wrapper of `pySHACL`_ to
+validate an ontology graph against the default Brick Schema constraints (called *shapes*) and user-defined shapes.
+
+.. _`pySHACL`: https://github.com/RDFLib/pySHACL
+"""
+import os
+import logging
+from rdflib import Graph, Namespace, URIRef, BNode, Literal
+from rdflib.plugins.sparql import prepareQuery
+from .namespaces import BRICK, A, RDF, RDFS, BRICK, BSH, SH, SKOS
+import pyshacl
+import io
+import pkgutil
+
+logging.basicConfig(
+    format="%(asctime)s,%(msecs)03d %(levelname)-7s [%(filename)s:%(lineno)d] %(message)s",
+    datefmt="%Y-%m-%d:%H:%M:%S",
+    level=logging.WARNING)
+
+class Validator():
+    """
+    Validates a data graph against Brick Schema and basic SHACL constraints for Brick.  Allows extra
+    constraints specific to the user's ontology.
+    """
+
+    # build accumulative namespace index from participating files
+    # build list of violations, each is a graph
+    def __init__(self, useBrickSchema=True, useDefaultShapes=True):
+        self.log = logging.getLogger('validate')
+        self.log.setLevel(logging.DEBUG if 'PYTEST_CURRENT_TEST' in os.environ else logging.WARNING)
+
+        self.log.info('Validate init')
+
```

Comment:
Logging the initialization should be done at the end of the `__init__` function. I think this also wants to be a `debug` too -- it's not crucial output.

Reply:
Sure.